src/overlays/actors/ovl_En_Col_Man/z_en_col_man.c

```diff
@@ -39,34 +61,220 @@ const ActorInit En_Col_Man_InitVars = {
+}
+
+void EnColMan_SetHeartPieceCollectedAndKill(EnColMan* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 6 && func_80147624(globalCtx)) {
+        gSaveContext.weekEventReg[0x38] |= 2;
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_80AFDF60(EnColMan* this) {
+    this->actor.draw = func_80AFE584;
+    this->actor.flags |= 0x10;
+    this->actor.flags |= 0x20;
+    this->unk148 = 2;
+    this->actionFunc = func_80AFDFB4;
+    this->actor.shape.shadowScale = 5.0f;
+    this->actor.gravity = -3.0f;
+}
+
+void func_80AFDFB4(EnColMan* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f velocity;
+    Vec3f accel;
+
+    this->scale = (BREG(55) / 10000.0f) + 0.0015f;
+
+    if ((this->actor.bgCheckFlags & 1) && (this->actor.velocity.y < 0.0f)) {
+        if (!this->hasSetRandomValues) {
+            this->actor.world.rot.y = randPlusMinusPoint5Scaled(30000.0f);
+            this->actor.speedXZ = 2.0f + BREG(56) + Rand_ZeroFloat(2.0f);
+            this->actor.velocity.y = 12.0f + BREG(57) + Rand_ZeroFloat(5.0f);
+            this->hasSetRandomValues = true;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANSATSUSYA_ROCK);
+            return;
+        }
+
+        for (i = 0; i < 2; i++) {
+            velocity.x = randPlusMinusPoint5Scaled(2.0f);
+            velocity.y = Rand_ZeroFloat(2.0f) + 1.0f;
+            velocity.z = randPlusMinusPoint5Scaled(2.0f);
+            accel.y = -0.1f;
+            accel.z = 0.0f;
+            accel.x = 0.0f;
 
-extern ColliderCylinderInit D_80AFE730;
+            func_800B0EB0(globalCtx, &this->actor.world.pos, &velocity, &accel, &primColor, &envColor,
+                          Rand_ZeroFloat(50.0f) + 60.0f, 0x1E, Rand_ZeroFloat(5.0f) + 20.0f);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Col_Man/EnColMan_Init.s")
+        Actor_MarkForDeath(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Col_Man/EnColMan_Destroy.s")
+void func_80AFE234(EnColMan* this) {
+    this->actor.draw = func_80AFE650;
+    this->unk148 = 3;
+    this->actionFunc = func_80AFE25C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Col_Man/func_80AFDD60.s")
+void func_80AFE25C(EnColMan* this, GlobalContext* globalCtx) {
+    this->scale = BREG(55) * 0.01f + 0.05f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Col_Man/func_80AFDE00.s")
+    if (BREG(60) || (this->actor.world.rot.z != 0)) {
+        if (this->actor.params == EN_COL_MAN_BOMB) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, this->actor.parent->world.pos.x,
+                        this->actor.parent->world.pos.y, this->actor.parent->world.pos.z, 0, 0, 0, 0);
+        } else {
+            EnBom* bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x,
+                                              this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0);
```

Comment:
This got me interested so I did some digging in KZ. It looks like what you have as `EN_COL_MAN_BOMB` is the "bomb" used by the Garo Master in stone tower (note it spawns an 'ACTOR_EN_CLEAR_TAG` so actually doesn't affect Link). `EN_COL_MAN_UNK4` are the bombs used by the other Garo Masters (On the moon and in secret shrine). These are real bombs that can hurt Link.

Reply:
Ah nice, I knew something was weird about it. I spawned in both using kz and didn't see much of a difference and didn't look far enough into it